protocol/api/api_versions_response.go

```diff
@@ -9,8 +9,8 @@ import (
 	"github.com/codecrafters-io/tester-utils/logger"
 )
 
-// ApiVersionsResponseKey contains the APIs supported by the broker.
-type ApiVersionsResponseKey struct {
+// ApiKeyVersionSupport contains the APIs supported by the broker.
+type ApiKeyVersionSupport struct {
```

Comment:
```suggestion
type ApiKeyEntry struct {
```

(or `ApiVersionsResponseApiKeyEntry`, or `ApiKeyListItem`)

The type here doesn't make it obvious that it's part of the `ApiVersionsResponse` body, and also introduces a new term ("Support") that doesn't have precise meaning